internal/inspect/inspect.go

```diff
@@ -0,0 +1,159 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package inspect
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+
+	"github.com/cartesi/rollups-node/internal/node/advancer/machines"
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/cartesi/rollups-node/internal/nodemachine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+var (
+	ErrInvalidMachines   = errors.New("machines must not be nil")
+	ErrInvalidRepository = errors.New("repository must not be nil")
+
+	ErrNoApp      = errors.New("no machine for application")
+	ErrBadRequest = errors.New("inspect bad request")
+)
+
+type Inspect struct {
+	machines Machines
+}
+
+type InspectResponse struct {
+	Status     string   `json:"status"`
+	Exception  string   `json:"exception"`
+	Reports    []string `json:"reports"`
+	InputIndex uint64   `json:"processed_input_count"`
+}
+
+// New instantiates a new Inspect.
+func New(machines Machines) (*Inspect, error) {
+	if machines == nil {
+		return nil, ErrInvalidMachines
+	}
+
+	return &Inspect{machines: machines}, nil
+}
+
+func (inspect *Inspect) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var (
+		dapp    Address
+		payload []byte
+		err     error
+		reports []string
+		status  string
+	)
+
+	if r.PathValue("dapp") == "" {
```

Comment:
I agree conceptually, however, this design came to be because I'm trying to emulate the previous API as close as possible, as requested during meetings.

The main problem is with multi dapp, I need to receive both the application address and the payload.

The previous API received an octet-stream on the post requests, so it doesn't make sense to receive any information that's not just the payload bytes.

Therefore, I arrived at this: receive the dapp address on path always, and payload on the path in get requests and and a binary on the body in post requests.

Reply:
I think it is safe to break the old design in this case.